Mage.Sets/src/mage/cards/k/KjeldoranGuard.java

```diff
@@ -0,0 +1,149 @@
+        );
+        ability.addTarget(new TargetCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private KjeldoranGuard(final KjeldoranGuard card) { super(card); }
+
+    @Override
+    public Card copy() {
+        return new KjeldoranGuard(this);
+    }
+}
+
+class KjeldoranGuardEffect extends OneShotEffect {
+
+    KjeldoranGuardEffect() {
+        super(Outcome.BoostCreature);
+        staticText = "Target creature gets +1/+1 until end of turn."
+                + "When that creature leaves the battlefield this turn, sacrifice {this}."
+                + "Activate only during combat and only if defending player controls no snow lands.";
+    }
+
+    private KjeldoranGuardEffect(KjeldoranGuardEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public KjeldoranGuardEffect copy() {
+        return new KjeldoranGuardEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID targetId = source.getFirstTarget();
+        Permanent targetPermanent = game.getPermanent(targetId);
+        if (game.getPermanent(targetId) == null) {
+            return false;
+        }
+
+        // Target creature gets +1/+1 until end of turn.
+        BoostTargetEffect buffEffect = new BoostTargetEffect(1, 1, Duration.EndOfTurn);
+        buffEffect.setTargetPointer(new FixedTarget(targetId, game));
+        game.addEffect(buffEffect, source);
+
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Guard.
+        game.addDelayedTriggeredAbility(
+                new KjeldoranGuardDelayedTriggeredAbility(
+                        new MageObjectReference(targetPermanent, game)
+                ), source
+        );
+
+        return true;
+    }
+}
+
+class KjeldoranGuardDelayedTriggeredAbility extends DelayedTriggeredAbility {
+
+    private final MageObjectReference mor;
+
+    KjeldoranGuardDelayedTriggeredAbility(MageObjectReference mor) {
+        super(new SacrificeSourceEffect(), Duration.EndOfTurn, true);
+        this.mor = mor;
+    }
+
+    private KjeldoranGuardDelayedTriggeredAbility(final KjeldoranGuardDelayedTriggeredAbility ability) {
+        super(ability);
+        this.mor = ability.mor;
+    }
+
+    @Override
+    public KjeldoranGuardDelayedTriggeredAbility copy() {
+        return new KjeldoranGuardDelayedTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent permanentLeaving = game.getPermanentOrLKIBattlefield(event.getTargetId());
+        return permanentLeaving != null && mor.equals(new MageObjectReference(permanentLeaving, game));
+    }
+
+    @Override
+    public String getRule() { return "sacrifice {this}"; }
```

Comment:
Hm, this doesn't feel quite right but not sure why, can you check the delayed trigger displays with correct text on the stack?

Reply:
I'm not sure we are displaying the trigger when those delayed trigger happens, is that was you feel is not right?

![image](https://github.com/magefree/mage/assets/34709007/9def22f2-7700-4fd3-b000-155e6184417b)

Reply:
will capitalize

Reply:
Yeah ideally the trigger phrase should be part of it (but for sure lots of these have wrong text all over the codebase). So not required to fix, just a nice to have.

Also it doesn't get automatically capitalized which is a separate bug.

Reply:
I can replace it with "When this creature leaves the battlefield, sacrifice Kjeldoran Guard"?

Reply:
Should be "When that creature leaves the battlefield this turn, sacrifice {this}."

Reply:
Just for a discussion point, got a comment from theelk801 last month that reflexive triggers can have their trigger left out. https://github.com/magefree/mage/pull/10866#discussion_r1298465762
So is that the line? Delayed should have trigger text but not reflexive ones? Or should that be a standalone dev discussion outside of that poor Ice Age common implementation.

Reply:
That's a good question - my initial reaction is I disagree but I should double check how it works.

Reply:
It's more important to have clarity on delayed, the reflexive at least occur in close proximity to their source